### PR TITLE
adding graph_coloring.cpp

### PR DESCRIPTION
**Pull request by sharyudongre**
Signed-off-by: sharyu <sharyudongre@gmail.com>

#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1301"></a>

### CODE REVIEW

File: graph/NP_complete_graph_coloring.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+and assigns each vertex its first available color. 
+*/
+
+// A C++ program to implement greedy algorithm for graph coloring 
+
+#include <iostream> 
+#include <list> 
+
+// A class that represents an undirected graph 
+
```
Comment (ridhishjain) [inline graph/NP_complete_graph_coloring.cpp:13]: add namespace

File: graph/NP_complete_graph_coloring.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+class Graph 
+{ 
+	int V;   // No. of vertices 
+	std::list<int> *adj;  // A dynamic array of adjacency lists 
+public:
+
+	// Constructor and destructor 
+	
+	Graph(int V) { this->V = V; adj = new std::list<int>[V]; } 
+	~Graph()	 { delete [] adj; }
```
Comment (ridhishjain) [inline graph/NP_complete_graph_coloring.cpp:23]: if the destructor is not used, remove it

File: graph/NP_complete_graph_coloring.cpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	int result[V]; 
+
+	// Assign the first color to first vertex 
+	
+	result[0] = 0; 
+
+	// Initialize remaining V-1 vertices as unassigned 
+	
+	for (int u = 1; u < V; u++) 
+		result[u] = -1;  // no color is assigned to u
```
Comment (ridhishjain) [inline graph/NP_complete_graph_coloring.cpp:54]: put it inside braces

File: graph/NP_complete_graph_coloring.cpp [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	for (int u = 1; u < V; u++) 
+		result[u] = -1;  // no color is assigned to u 
+
+	// A temporary array to store the available colors. True 
+	// value of available[color] would mean that the color 'color' is 
+	// assigned to one of its adjacent vertices 
+	
+	bool available[V]; 
+	for (int color = 0; color < V; color++) 
+		available[color] = false;
```
Comment (ridhishjain) [inline graph/NP_complete_graph_coloring.cpp:62]: curly braces

File: graph/NP_complete_graph_coloring.cpp [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+		
+		std::list<int>::iterator i; 
+		for (i = adj[u].begin(); i != adj[u].end(); ++i) 
+			if (result[*i] != -1) 
+				available[result[*i]] = true; 
+
+		// Find the first available color 
+		
+		int color; 
+		for (color = 0; color < V; color++)
```
Comment (ridhishjain) [inline graph/NP_complete_graph_coloring.cpp:79]: write int inside for loop and use braces

File: graph/NP_complete_graph_coloring.cpp [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+		for (i = adj[u].begin(); i != adj[u].end(); ++i) 
+			if (result[*i] != -1) 
+				available[result[*i]] = true; 
+
+		// Find the first available color 
+		
+		int color; 
+		for (color = 0; color < V; color++) 
+			if (available[color] == false) 
+				break;
```
Comment (ridhishjain) [inline graph/NP_complete_graph_coloring.cpp:81]: curly braces

File: graph/NP_complete_graph_coloring.cpp [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+			if (result[*i] != -1) 
+				available[result[*i]] = false; 
+	} 
+
+	// print the result 
+	
+	for (int u = 0; u < V; u++) 
+		std::cout << "Vertex " << u << " ---> Color "
+			<< result[u] << std::endl; 
+}
```
Comment (ridhishjain) [inline graph/NP_complete_graph_coloring.cpp:97]: curly braces

**Review by ridhishjain — CHANGES_REQUESTED** [review]
please change these things

**Comment by Panquesito7** [comment]
Isn't this a duplicate of [`backtracking/graph_coloring.cpp`](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/backtracking/graph_coloring.cpp)?

**Comment by sharyudongre** [comment]
> Isn't this a duplicate of [`backtracking/graph_coloring.cpp`](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/backtracking/graph_coloring.cpp)?

this is by using greedy algorithm.

**Pull request closed without merging**